packages/dev/docs/src/client.js

```diff
@@ -183,8 +185,100 @@ function Hamburger() {
   );
 }
 
+function DocSearch() {
+  const client = algoliasearch('1V1Q59JVTR', '44a7e2e7508ff185f25ac64c0a675f98');
+  const searchIndex = client.initIndex('react-spectrum');
+  const searchOptions = {
+    highlightPreTag: `<mark class="${docsStyle.docSearchBoxMark}">`,
+    highlightPostTag: '</mark>'
+  };
+
+  const [searchValue, setSearchValue] = useState('');
+  const [predictions, setPredictions] = useState(null);
+  const [suggestions, setSuggestions] = useState(null);
+
+  let updatePredictions = ({hits}) => {
+    setPredictions(hits);
+    let sections = [];
+    hits.forEach(prediction => {
+      let hierarchy = prediction.hierarchy;
+      let objectID = prediction.objectID;
+      let lvl0 = hierarchy.lvl0;
+      let section = sections.find(section => section.title === lvl0);
+      if (!section) {
+        section = {title: lvl0, items: []};
+        sections.push(section);
+      }
+      let text = [];
+      let textValue = [];
+      for (let i = 1; i < 6; i++) {
+        if (hierarchy[`lvl${i}`]) {
+          text.push(prediction._highlightResult.hierarchy[`lvl${i}`].value);
+          textValue.push(hierarchy[`lvl${i}`]);
+        }
+      }
+      section.items.push(
+        <Item key={objectID} textValue={textValue.join(' | ')}>
+          <Text><span dangerouslySetInnerHTML={{__html: text.join(' | ')}} /></Text>
+          {
+            prediction.content &&
+            <Text slot="description">
+              <span dangerouslySetInnerHTML={{__html: prediction._snippetResult.content.value}} />
+            </Text>
+          }
+        </Item>
+      );
+    });
+    let suggestions = sections.map((section, index) => <Section key={`${index}-lvl0-${section.title}`} title={section.title}>{section.items}</Section>);
```

Comment:
Currently, I just see a `<Section>` for "Documentation". It might be nice to have sections that match the site's sections, like React Spectrum, React Aria, React Stately, Releases, Blog, etc. so its organized a bit (if that isn't too difficult to do with this setup)

Reply:
I think that is something that can be managed through the AlgoliaSearch index setup. I just went with what had been set up already, but I can try to refine the categories.

Reply:
See 42e258bf524db9531272964579e9564c22a8ff47